### DISCUSSION TITLE
Support "stack" filtering for the Awesome list

### DISCUSSION THREAD

**Discussion by Daenyth** [scalacenter/scaladex#943] [Ideas]
I'd find it really helpful if the new "awesome" list could filter for stack. A really common use case is "I want to find a library for X in the lightbend|typelevel|zio|haoyiverse ecosystem", in order to minimize friction between different stack integrations.

One way to implement this might be to have the tool look at the library dependencies. If it depends on cats-effect for example, it would be listed in the typelevel stack

**Comment by adpi2** [comment]
Thanks @Daenyth, I like the idea!

We already have a notion of `ProjectDependency`: project A depends on project B if there is one artifact in project A that depends on one artifact of project B. We can reuse it to create the notion of stack.

In terms of implementation we can add a field `dependencies: Seq[Project.Reference]` in the data class [`ProjectDocument`](https://github.com/scalacenter/scaladex/blob/main/modules/core/shared/src/main/scala/scaladex/core/model/search/ProjectDocument.scala) that is indexed in elasticsearch and then filter on it in the elasticsearch request, depending on the stack parameter.

Would you be interested in implementing that feature?

**Reply by Daenyth** [reply]
I don't have the time/energy to commit to that unfortunately

**Reply by adpi2** [reply]
ok no problem
